hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieWriteHandle.java

```diff
@@ -140,6 +143,8 @@ protected HoodieWriteHandle(HoodieWriteConfig config, String instantTime, String
         && hoodieTable.getMetaClient().getTableConfig().getRecordMergeMode() == EVENT_TIME_ORDERING
         && ConfigUtils.isTrackingEventTimeWatermark(config.getProps());
     this.keepConsistentLogicalTimestamp = isTrackingEventTimeWatermark && ConfigUtils.shouldKeepConsistentLogicalTimestamp(config.getProps());
+    this.deleteContext = new DeleteContext(ConfigUtils.getMergeProps(config.getProps(), hoodieTable.getMetaClient().getTableConfig().getProps()), writeSchema).withReaderSchema(writeSchema);
+    this.deleteContextWithMetaFields = new DeleteContext(ConfigUtils.getMergeProps(config.getProps(), hoodieTable.getMetaClient().getTableConfig().getProps()), writeSchemaWithMetaFields).withReaderSchema(writeSchemaWithMetaFields);
```

Comment:
the meta fields are not there for in-memory records in write path, so the `DeleteContext#withReaderSchema` does not really work. So just maintain one singleton `DeleteContext` in the write handle should be fine, can use the flag `preserveMetadata` to decide which schema to use.

Reply:
yes, we might need just one of the deleteContext for one instance of a `HoodieWriteHandle`.

Reply:
sorry, I meant for single delete context per write handle.